src/code-components/RichTextArea/RichTextArea.tsx

```diff
@@ -0,0 +1,77 @@
+import { useMemo, Suspense, lazy, ReactNode } from "react";
+import { Range } from "quill/core";
+import {
+  formatDefaultToolbarConfigs,
+  Toolbar,
+  ToolbarConfigs,
+} from "./formatDefaultToolbarConfigs";
+
+const Editor = lazy(() =>
+  import("./Editor").then((module) => ({ default: module.Editor })),
+);
+
+interface RichTextAreaProps {
+  htmlValue?: string;
+  toolbar: Toolbar;
+  customToolbar?: ToolbarConfigs;
+  onSelectionChange?: (range: Range | null, source: string) => void;
+  onChange?: (content: string, source: string) => void;
+  onBlur?: (range: Range | null, source: string) => void;
+  onFocus?: (range: Range | null, source: string) => void;
+  onKeyDown?: (event: KeyboardEvent) => void;
+  onKeyUp?: (event: KeyboardEvent) => void;
+  placeholder?: string;
+  readOnly: boolean;
+  wrapperClassName: string;
+  ariaLabel?: string;
+  ariaLabeledby?: string;
+  fallback: ReactNode;
+}
+
+export function RichTextArea({
+  htmlValue,
+  toolbar,
+  customToolbar,
+  onSelectionChange,
+  onChange,
+  onBlur,
+  onFocus,
+  onKeyDown,
+  onKeyUp,
+  placeholder,
+  readOnly,
+  wrapperClassName,
+  ariaLabel,
+  ariaLabeledby,
+  fallback,
+}: RichTextAreaProps) {
+  const formattedToolbar = useMemo(
+    () => formatDefaultToolbarConfigs(toolbar),
+    [JSON.stringify(toolbar)],
+  );
+  const currentToolbarConfigs = customToolbar?.length
+    ? customToolbar
+    : formattedToolbar.length
+      ? formattedToolbar
+      : false;
+
+  return (
+    <Suspense fallback={fallback}>
```

Comment:
Do we really need to complicate this component (and all its' parents) with a fallback?

Why don't we treat quill as a simple `<textarea`? that is, render it always, no matter if we're on a client or a server?